com.unity.render-pipelines.core/Runtime/Debugging/DebugUI.Containers.cs

```diff
@@ -53,14 +55,19 @@ public Container(string displayName, ObservableList<Widget> children)
                 this.children = children;
                 children.ItemAdded += OnItemAdded;
                 children.ItemRemoved += OnItemRemoved;
+
+                // Call OnAdded callback for already existing items to ensure their panel & parent are set
+                for (int i = 0; i < this.children.Count; i++)
+                    OnItemAdded(this.children, new ListChangedEventArgs<Widget>(i, this.children[i]));
             }
 
             internal override void GenerateQueryPath()
             {
                 base.GenerateQueryPath();
 
-                foreach (var child in children)
-                    child.GenerateQueryPath();
+                int numChildren = children.Count;
```

Comment:
I see this pattern being changed in several places here. I'm curious, why is this needed?

Reply:
This is an optimization. `UnityEngine.Rendering.ObservableList` implementation is such that using foreach to iterate causes allocations. I wanted to get rid of them, so I removed the usage of foreach. A better fix would be to improve the implementation of ObservableList in this respect, but I didn't want to get into it in this PR.